tests/spark/test_spark_connect_model_export.py

```diff
@@ -0,0 +1,146 @@
+import json
+import os
+
+import numpy as np
+import pandas as pd
+import pyspark
+import pytest
+from packaging.version import Version
+
+if Version(pyspark.__version__) < Version("3.5"):
+    pytest.skip("pyspark ML connect Model is available in pyspark >= 3.5")
+
+from pyspark.ml.connect.classification import LogisticRegression
+from pyspark.ml.connect.feature import StandardScaler
+from pyspark.ml.connect.pipeline import Pipeline
+from pyspark.sql import SparkSession
+from pyspark.sql import functions as spark_f
+from pyspark.sql.types import LongType
+from sklearn import datasets
+
+import mlflow
+import mlflow.pyfunc.scoring_server as pyfunc_scoring_server
+from mlflow import pyfunc
+from mlflow.pyfunc import spark_udf
+
+from tests.helper_functions import (
+    pyfunc_serve_and_score_model,
+)
+from tests.pyfunc.test_spark import score_model_as_udf
+from tests.spark.test_spark_model_export import (
+    SparkModelWithData,
+)
+
+
+def _get_spark_connect_session():
+    return (
+        SparkSession.builder.remote("local[2]")
+        .config("spark.connect.copyFromLocalToFs.allowDestLocal", "true")
+        .getOrCreate()
+    )
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
+
+
+def score_model_as_udf(model_uri, pandas_df, result_type):
+    spark = SparkSession.getActiveSession()
+    spark_df = spark.createDataFrame(pandas_df).coalesce(1)
+    pyfunc_udf = spark_udf(
+        spark=spark, model_uri=model_uri, result_type=result_type, env_manager="local"
+    )
+    new_df = spark_df.withColumn("prediction", pyfunc_udf(spark_f.struct(spark_f.col("features"))))
+    return new_df.toPandas()["prediction"]
+
+
+@pytest.fixture(scope="module")
+def spark():
+    spark = _get_spark_connect_session()
+    yield spark
+    spark.stop()
+
+
+@pytest.fixture(scope="module")
+def iris_df(spark):
+    X, y = datasets.load_iris(return_X_y=True)
+    spark_df = spark.createDataFrame(zip(X, y), schema="features: array<double>, label: long")
+    return spark_df.toPandas(), spark_df
+
+
+@pytest.fixture(scope="module")
+def spark_model(iris_df):
+    iris_pandas_df, iris_spark_df = iris_df
+    scaler = StandardScaler(inputCol="features", outputCol="scaled_features")
+    lr = LogisticRegression(maxIter=50, numTrainWorkers=2, learningRate=0.001)
```

Comment:
```suggestion
    lr = LogisticRegression(maxIter=10, numTrainWorkers=2, learningRate=0.001)
```

for faster training.

Reply:
I guess the difference is subtle but 50 sounds too large for testing.